scripts/docker_build

```diff
@@ -11,7 +11,9 @@ source "$SCRIPTS_DIR/common"
 # https://github.com/googleapis/google-cloud-cpp/blob/a186208b79d900b4ec71c6f9df3acf7638f01dc6/ci/kokoro/docker/build.sh#L147-L152
 readonly DOCKER_UID="${UID:-0}"
 readonly DOCKER_GID="$(id -g)"
-readonly DOCKER_USER="${USER:-root}"
+# We use the `oak` Docker user in order to maintain library paths on different
+# machines and to make Wasm modules reproducible.
+readonly DOCKER_USER="oak"
```

Comment:
Maybe `'docker'` would be a more common value for the username.

Reply:
In fact, why keep this variable if it's always constant? I think we can remove this and also the corresponding ARG from the Docker image, right?

Reply:
Agree
But why removing the ARG from Dockerfile? We could just change it to `ARG USERNAME=docker`
https://github.com/project-oak/oak/blob/68b4de822f56b40beab4d498e3a778ff98940191/Dockerfile#L286-L295

Reply:
Yeah makes sense, it is already used in a couple of places.

Reply:
Done